spyder/app/tour.py

```diff
@@ -510,14 +511,23 @@ def update_canvas(self):
             self.path_current = self.path_full
 
         if self.decoration is not None:
-            for widget in self.decoration:
+            for widgets in self.decoration:
+                if isinstance(widgets, QWidget):
+                    widgets = [widgets]
+                geoms = []
+                for widget in widgets:
+                    widget.raise_()
+                    widget.show()
+                    geo = widget.frameGeometry()
+                    width, height = geo.width(), geo.height()
+                    point = widget.mapTo(self.parent, QPoint(0, 0))
+                    x, y = point.x(), point.y()
+                    geoms.append((x, y, width, height))
+                x = min([geom[0] for geom in geoms])
+                y = min([geom[1] for geom in geoms])
+                width = max([geom[0]+geom[2] for geom in geoms])-x
+                height = max([geom[1]+geom[3] for geom in geoms])-y

```

Comment:
```suggestion
                width = max([geom[0] + geom[2] for geom in geoms]) - x
                height = max([geom[1] + geom[3] for geom in geoms]) - y
```

Code is easier to read with spaces around math signs (even if it takes more than 79 columns).

Reply:
The line can be broken before the `for` to avoid PEP 8 issues